dashboard/src/components/DeploymentForm/DeploymentForm.tsx

```diff
@@ -43,12 +43,18 @@ export interface IDeploymentFormState {
 class DeploymentForm extends React.Component<IDeploymentFormProps, IDeploymentFormState> {
   public state: IDeploymentFormState = {
     releaseName: Moniker.choose(),
-    appValues: "",
+    appValues: this.props.selected.values || "",
     isDeploying: false,
     latestSubmittedReleaseName: "",
     valuesModified: false,
   };
 
+  public componentWillReceiveProps(nextProps: IDeploymentFormProps) {
+    if (nextProps.selected.version !== this.props.selected.version && !this.state.valuesModified) {
+      this.setState({ appValues: nextProps.selected.values || "" });
+    }
+  }
```

Comment:
Again, this looks like we're adding more code which calculates state by detecting changes in props, rather than simply a function of the state... I worry we're going to pay for this later when `componentWillReceiveProps` is no longer supported (it'll be removed in 1.17).

Can we first update to React 16.3 (at least, but possibly latest - 16.11), before we add more code like this? At least then we'll be able to use `getDerivedStateFromProps` and will be forced to add `UNSAFE_` to these unsafe methods we're adding?

Reply:
Sounds good.